plugins/azure/fix_plugin_azure/resource/postgresql.py

```diff
@@ -0,0 +1,641 @@
+    supported_ha_mode: Optional[List[str]] = field(default=None, metadata={'description': 'Supported high availability mode'})  # fmt: skip
+    supported_hyperscale_node_editions: Optional[List[AzureHyperscaleNodeEditionCapability]] = field(default=None, metadata={'description': ''})  # fmt: skip
+    capability_zone: Optional[str] = field(default=None, metadata={"description": "zone name"})
+    zone_redundant_ha_and_geo_backup_supported: Optional[bool] = field(default=None, metadata={'description': 'A value indicating whether a new server in this region can have geo-backups to paired region.'})  # fmt: skip
+    zone_redundant_ha_supported: Optional[bool] = field(default=None, metadata={'description': 'A value indicating whether a new server in this region can support multi zone HA.'})  # fmt: skip
+    location: Optional[str] = field(default=None, metadata={'description': 'The geo-location where the resource lives'})  # fmt: skip
+
+    def post_process(self, graph_builder: GraphBuilder, source: Json) -> None:
+
+        def collect_editions(edition: AzureFlexibleServerEditionCapability) -> None:
+            server_types = []
+            for version in edition.supported_server_versions or []:
+                for sku in version.supported_vcores or []:
+                    for supported_storage in edition.supported_storage_editions or []:
+                        for storage in supported_storage.supported_storage_mb or []:
+                            server_type = {
+                                "id": f"{edition.name}_{version.name}_{sku.name}_{storage.name}",
+                                "name": f"{edition.name}_{version.name}_{sku.name}_{storage.name}",
+                                "sku": {
+                                    "name": sku.name,
+                                    "tier": edition.name,
+                                    "vCores": sku.v_cores,
+                                    "memoryPerVCoreMb": sku.supported_memory_per_vcore_mb,
+                                },
+                                "storage": {
+                                    "iops": storage.supported_iops,
+                                    "storageSizeGb": (
+                                        storage.storage_size_mb // 1024 if storage.storage_size_mb else 0
+                                    ),
+                                },
+                                "location": self.location,
+                            }
+                            server_types.append(server_type)
+
+            AzurePostgresqlServerType.collect(server_types, graph_builder)
+
+        if self.supported_psql_flexible_server_editions:
+            for edition in self.supported_psql_flexible_server_editions:
+                graph_builder.submit_work(service_name, collect_editions, edition)
+
+
+@define(eq=False, slots=False)
+class AzurePostgresqlServerConfiguration(MicrosoftResource, AzureProxyResource):
```

Comment:
Same comment as on MySQL:
I would imagine something like this:

```python
class AzureDatabseConfiguration(MicrosoftResource):
   config: Optional[Json] = None
```

We would  read all ServerConfiguration Resources and turn them into a simple dict.